libsolidity/parsing/Parser.cpp

```diff
@@ -1981,6 +1986,26 @@ pair<vector<ASTPointer<Expression>>, vector<ASTPointer<ASTString>>> Parser::pars
 	return ret;
 }
 
+optional<string> Parser::findLicenseString()
+{
+	// We circumvent the scanner here, because it skips non-docstring comments.
+	static regex const licenseRegex("SPDX-License-Identifier:\\s+([^\\n\\r]+)");
+
+	smatch match;
+	if (regex_search(m_scanner->source(), match, licenseRegex))
+		return boost::trim_copy(string(match[1]));
+
+	parserWarning(
+		0_error,
+		{-1, -1, m_scanner->charStream()},
+		"SPDX license identifier not provided in source file. "
+		"Before publishing, consider adding a comment containing "
+		"\"SPDX-License-Identifier: <SPDX-License>\" to each source file. "
```

Comment:
It is still unclear what to use for closed source:
a) "Proprietary" perhaps?
b) avoiding the field?

Reply:
Provide the field with `UNLICENSED` or `CLOSED-SOURCE` or `PROPRIETARY` - do we need to specify that?

Reply:
I think we should provide a recommendation given SPDX does not provide one.

Reply:
npm proposes `UNLICENSED`: https://docs.npmjs.com/files/package.json#license

Reply:
Sounds good, we should document this and link to npm as well.